packages/sdk/client/src/echo/space-proxy.ts

```diff
@@ -412,8 +414,16 @@ export class SpaceProxy implements Space {
     };
   }
 
-  private async _createEpoch({ migration }: { migration?: CreateEpochRequest.Migration } = {}) {
-    await this._clientServices.services.SpacesService!.createEpoch({ spaceKey: this.key, migration });
+  private async _createEpoch({
+    migration,
+    automergeRootUrl,
+  }: { migration?: CreateEpochRequest.Migration; automergeRootUrl?: string } = {}) {
+    log('create epoch', { migration, automergeRootUrl });
+    await this._clientServices.services.SpacesService!.createEpoch({ spaceKey: this.key, migration, automergeRootUrl });
+    if (this._db.rootUrl !== automergeRootUrl) {
+      this._databaseInitialized.reset();
```

Comment:
I dont think this is a good choice since database does not become un-initialized while the epoch is chainging.

A better solution would be an event (maybe on the DB iteself?) to watch for root changes